app/src/main/java/com/xa/xpensauditor/TabFragment.java

```diff
@@ -89,28 +108,51 @@ public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
         arrayAdapterTF=new ArrayAdapter<String>(view.getContext(),android.R.layout.simple_list_item_1,CatgTF);
 
         RefCat.addChildEventListener(new ChildEventListener() {
+            /**
+             * Triggered when a child node is added.
+             * @param dataSnapshot
+             * @param s
+             */
             @Override
             public void onChildAdded(DataSnapshot dataSnapshot, String s) {
                 String value= dataSnapshot.getKey().trim();
                 CatgTF.add(value);
                 arrayAdapterTF.notifyDataSetChanged();
             }
 
+            /**
+             * Triggered when a child node in the database is updated or added.
+             * @param dataSnapshot
+             * @param s
+             */
             @Override
             public void onChildChanged(DataSnapshot dataSnapshot, String s) {
 
             }
 
+            /**
+             * Triggered when a child node in the database is removed.
+             * @param dataSnapshot
+             */
             @Override
             public void onChildRemoved(DataSnapshot dataSnapshot) {
 
             }
 
+            /**
+             * Triggered when a child node in the database is moved.
+             * @param dataSnapshot
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.NonEmptyAtclauseDescriptionCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>At-clause should have a non-empty description.